Zend/zend_alloc.c

```diff
@@ -663,35 +696,47 @@ static zend_always_inline int zend_mm_bitset_is_free_range(zend_mm_bitset *bitse
 
 static void *zend_mm_chunk_alloc_int(size_t size, size_t alignment)
 {
-	void *ptr = zend_mm_mmap(size);
-
+#ifdef _WIN32
+	void *ptr = NULL;
+	size_t offset;
+#ifndef _WIN64
+	/* 32-bit build has limited memory address space and partical deallocation is not supported,
+	 * try to reserve exact aligned size first, GH-9650 */
+	ptr = zend_mm_mmap_reserve(size + alignment);
```

Comment:
This is new code. I think coding style guidelines allowed for declaring variables after other executed statements for php 8.0 and even older (no need to support old C standards)

`void *ptr = zend_mm_mmap_reserve(size + alignment);`

Reply:
`#ifndef _WIN64` there is to detect non-x64, ie. x86 only, thus declaration above is needed